backend/src/paymaster/index.ts

```diff
@@ -1124,12 +1125,14 @@ export class Paymaster {
     const records = await coingeckoRepo.findAll();
     const tokenIds = records.map((record: { coinId: any; }) => record.coinId);
 
-    const data = await this.coingeckoService.fetchPriceByCoinID(tokenIds);
+    const data = await this.coingeckoService.fetchPriceByCoinID(tokenIds, log);
     const tokenPrices: any = [];
     records.map(record => {
       tokenPrices[ethers.utils.getAddress(record.address)] = { price: Number(data[record.coinId].usd).toFixed(5), decimals: record.decimals, gasToken: tokenAddress, symbol: record.token }
     })
-    const tokenData = tokenPrices[tokenAddress];
+    let tokenData = tokenPrices[tokenAddress];
+    if (!tokenData) tokenData = this.coingeckoPrice.get(cacheKey)?.data;
+    if (!tokenData) throw new Error(ErrorMessage.COINGECKO_PRICE_NOT_FETCHED);
```

Comment:
This thrown error should be logged.

Reply:
yes that is logged on the coingecko service if its errored out on coingecko request